start

```diff
@@ -502,4 +507,46 @@ popd () {
     command popd "$@" > /dev/null
 }
 
+function sorobanRpcStatus () {
+    horizonStatus
+    echo "waiting for soroban rpc to report ready state..."
+    COUNTER=1
+    while ! $(curl --silent --location --request POST 'http://localhost:8000/soroban/rpc' \
+                --header 'Content-Type: application/json' \
+        --data-raw '{
+          "jsonrpc": "2.0",
+          "id": 10235,
+          "method": "getHealth"
+          
+        }' | jq --exit-status '.result.status == "healthy"' 2>/dev/null | grep -o true || echo false ); 
+    do
+      if [ $COUNTER -gt 12 ]; then 
+          echo "waiting for soroban rpc to report ready state..."
+          COUNTER=1
+      else 
+          COUNTER=$[$COUNTER +1]    
+      fi 
+      sleep 5
+    done 
+    echo "soroban rpc reported ready status, the service can be used by tools/cli now ..."
+}
+
+function horizonStatus () {
```

Comment:
💡 It would be great to add a function that does the same for the stellar-core. The state of stellar-core is often what I am most interested in because for joining any network it is the state that takes the longest to resolve. Stellar-core has an `info` endpoint with a field that prints the latest status, and that status updates frequently, so printing it out as it changes would give a lot of really useful feedback to someone sitting there and waiting.

It would be awesome to poll the info endpoint, and if the status value is different to the previous value, to print it out. That way the user gets an updated list of what's going on and why it's taking 2 mins.

This doesn't have to happen in this PR, we can do it in a follow up and I'm happy to contribute it, but you're welcome to as well.

Reply:
I think core's real time status is also getting indirectly checked by this latest horizon status check:
https://github.com/stellar/quickstart/pull/403/files#diff-cced28c6dc3f99c2396a5eaad732bf6b28142335892b1cd0e6af6cdb53f5ccfaR539

I think horizon gets 'core_latest_version' from that same endpoint and the horizon check compares that ledger to horizon's latest ingested ledger to make sure the two are at least within 5 ledgers of each other, signaling that both services are progressing.

Reply:
My suggestion is not to check the status of core, it's to print the status of core. The status of core is much more informative than the information horizon exposes. Iirc core will tell us how the download is going, whether it is synced, etc.

Reply:
for tracking, created feature request - https://github.com/stellar/quickstart/issues/407